runtime/src/message_processor.rs

```diff
@@ -257,8 +221,38 @@ impl MessageProcessor {
         )
     }
 
-    fn sum_account_lamports(accounts: &mut [&mut Account]) -> u128 {
-        accounts.iter().map(|a| u128::from(a.lamports)).sum()
+    fn verify_account_references(
+        executable_accounts: &mut [(Pubkey, RefCell<Account>)],
+        program_accounts: &mut [Rc<RefCell<Account>>],
+    ) -> Result<(), InstructionError> {
+        for account in program_accounts.iter() {
+            account
+                .try_borrow_mut()
+                .map_err(|_| InstructionError::AccountBorrowOutstanding)?;
+        }
+        for (_, account) in executable_accounts.iter() {
+            account
+                .try_borrow_mut()
+                .map_err(|_| InstructionError::AccountBorrowOutstanding)?;
+        }
+        Ok(())
+    }
+
```

Comment:
I think this might have been broken before for accounts with identical member values, fixed now